### DISCUSSION TITLE
Behold that idea for a general purpose "meson run" command

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson#12915] [Ideas]
Ok hear me out just simply thinking a general perpuse run command would be an interesting idea, some possibilities of usage could be the following:

meson run "prog name"

meson run "prog name" -C builddir

**Comment by xclaesse** [comment]
What would it do?

**Reply by dreamer-coding** [reply]
runs a binary by name, and should be possible to select a builddir to run the named executable. Would remove the need to input a full path to an executable and or remember what the dot extinction was for the executable

**Reply by xclaesse** [reply]
That's already supported: `meson devenv -C builddir myapp`. See https://mesonbuild.com/Commands.html#devenv.

**Comment by eli-schwartz** [comment]
Nack to renaming `meson devenv`, which is the only practical difference I can see between devenv and the proposed "run".

**Reply by dreamer-coding** [reply]
Fair. I didn't even know that this was added so perhaps a name change to this command would make it more intuitive.
